source/include/shadow.h

```diff
@@ -92,12 +92,14 @@ typedef enum ShadowTopicStringType
  */
 typedef enum ShadowStatus
 {
-    SHADOW_SUCCESS = 0,                     /**< @brief Shadow function success. */
-    SHADOW_FAIL,                            /**< @brief Shadow function encountered error. */
-    SHADOW_BAD_PARAMETER,                   /**< @brief Input parameter is invalid. */
-    SHADOW_BUFFER_TOO_SMALL,                /**< @brief The provided buffer is too small. */
-    SHADOW_THINGNAME_PARSE_FAILED,          /**< @brief Could not parse the thing name. */
-    SHADOW_SHADOW_MESSAGE_TYPE_PARSE_FAILED /**< @brief Could not parse the shadow type. */
+    SHADOW_SUCCESS = 0,              /**< @brief Shadow function success. */
+    SHADOW_FAIL,                     /**< @brief Shadow function encountered error. */
+    SHADOW_BAD_PARAMETER,            /**< @brief Input parameter is invalid. */
+    SHADOW_BUFFER_TOO_SMALL,         /**< @brief The provided buffer is too small. */
+    SHADOW_THINGNAME_PARSE_FAILED,   /**< @brief Could not parse the thing name. */
+    SHADOW_ROOT_PARSE_FAILED,        /**< @brief Could not parse the classic or named shadow root. */
+    SHADOW_SHADOWNAME_PARSE_FAILED,  /**< @brief Could not parse the shadow name (in the case of a named shadow topic). */
```

Comment:
Although this is already a major revision of the library I soggiest moving new enums to the end to help a little bit backwards compatibility.

Reply:
Done